src/nunavut/lang/cpp/templates/_composite_type.j2

```diff
@@ -28,10 +28,7 @@
 // +-------------------------------------------------------------------------------------------------------------------+
 {% endifuses -%}
 {{ composite_type.doc | block_comment('cpp-doxygen', 0, 120) }}
-{% if composite_type.deprecated -%}
-[[deprecated("{{ composite_type }} is reaching the end of its life; there may be a newer version available")]]
-{% endif -%}
-struct {{composite_type|short_reference_name}} final
+struct {{"[[deprecated(\"{} is reaching the end of its life; there may be a newer version available\")]]".format(composite_type) if composite_type.deprecated else ""}} {{composite_type|short_reference_name}} final
```

Comment:
How about this

```suggestion
struct {% if composite_type.deprecated -%}
[[deprecated("{{ composite_type }} is reaching the end of its life; there may be a newer version available")]]
{%- endif -%}
{{composite_type|short_reference_name}} final
```

Reply:
Fine for me, is the whitespace after the struct needed?

Reply:
Yes, it is quite essential.